storagedriver/factory/factory.go

```diff
@@ -58,7 +58,12 @@ func Create(name string, parameters map[string]interface{}) (storagedriver.Stora
 		// }
 		// return driverClient, nil
 	}
-	return driverFactory.Create(parameters)
+	d, err := driverFactory.Create(parameters)
```

Comment:
You can wrap the driver in the testsuites.go like I did here? I am not sure why we need to move it to the New of each driver? If you want to always get a wrapped instance from New, then we should make all drivers constructable through one method by applying a creation pattern here. Final call is yours, that's not my code. But in my opinon when you introduce redundancy that linearly increases like this, it is a bad sign about the design.